build.sh

```diff
@@ -1,12 +1,15 @@
 #!/bin/bash
 set -e
-[ -z $EM_DIR] && EM_DIR=~/src/emscripten
+[ -z "$EM_DIR" ] && EM_DIR=~/src/emscripten
+
+OPTZ="-Oz -O3"
 
 do_config() {
     echo config
 # something wrong with emcc + cproto, use gcc as CPP instead
-CPPFLAGS="-Os -DFEAT_GUI_WEB" \
-CPP="gcc -E" \
+CPPFLAGS="$OPTZ -DFEAT_GUI_WEB" \
+CFLAGS="$OPTZ" \
+CPP="emcc -E" \
```

Comment:
Yes. I think it worked. But for emterpreter I had to use emscripten 1.29 for the configure step and the latest version for make and link.